core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java

```diff
@@ -298,7 +318,8 @@ private void toggleSelection(ImageView favicon, HistoryListItem.HistoryItem hist
     }
   }
 
-  @Override protected void injection(CoreComponent coreComponent) {
+  @Override
+  protected void injection(CoreComponent coreComponent) {
     coreComponent.inject(this);
```

Comment:
you will need to build a CoreActivityComponent here to inject a DialogShower. You will need to add an inject method in 
CoreActivityComponent and delete the old inject method in `CoreComponent`

Reply:
Following the above steps leads to a crash when I open the `History Activity`.

Log:
```
2020-02-27 18:31:31.588 23186-23186/org.kiwix.kiwixmobile E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.kiwix.kiwixmobile, PID: 23186
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.kiwix.kiwixmobile/org.kiwix.kiwixmobile.core.history.HistoryActivity}: java.lang.IllegalStateException: android.app.Activity must be set
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3123)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3266)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1957)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7099)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
     Caused by: java.lang.IllegalStateException: android.app.Activity must be set
        at dagger.internal.Preconditions.checkBuilderRequirement(Preconditions.java:95)
        at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent$CoreActivityComponentBuilder.build(DaggerCoreComponent.java:509)
        at org.kiwix.kiwixmobile.core.history.HistoryActivity.injection(HistoryActivity.java:318)
        at org.kiwix.kiwixmobile.core.base.BaseActivity.onCreate(BaseActivity.kt:43)
        at org.kiwix.kiwixmobile.core.history.HistoryActivity.onCreate(HistoryActivity.java:144)
        at android.app.Activity.performCreate(Activity.java:7327)
        at android.app.Activity.performCreate(Activity.java:7318)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1275)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3103)
```